StRoot/StSpinPool/StJetSkimEvent/StJetSkimEvent.h

```diff
@@ -145,357 +145,397 @@ inline void StJetSkimTrig::setTotalEnergy(int aEnergy) {mTotalEnergy = aEnergy;}
-		int mL2Result[64];
-
-		// For run 13 only saving a 4th JP threshold for the di-jet triggers
-		// For any other run this 4th space will be set to zero
-		// Care should be taken when calling the JP thresholds from the skim.root file 
-		// in any additional code
-		// Modified by Danny
-
-		//int mBarrelJetPatchTh[3];
-		//int mEndcapJetPatchTh[3];
-		int mBarrelJetPatchTh[4];
-		int mEndcapJetPatchTh[4];
-		int mOverlapJetPatchTh[3];
-
-		int mBarrelHighTowerTh[4];
-		int mEndcapHighTowerTh[2];
-
-		int mBarrelJetPatchAdc[18];
-		int mEndcapJetPatchAdc[6];
-		int mOverlapJetPatchAdc[6];
-
-		int mEmcLayer2;
-
-		TDatime mDatime;
-
-		ClassDef(StJetSkimEvent,7);
+public:
+  StJetSkimEvent();
+  StJetSkimEvent(const StJetSkimEvent &other);
+  virtual ~StJetSkimEvent();
+  StJetSkimEvent& operator=(const StJetSkimEvent &rhs);
+
+  //action
+  void clear();
+  void Clear(const char *option="");
+
+  //sets
+  void setTrig(const StJetSkimTrig& );
+  void setVert(const StJetSkimVert& );
+
+  void setTrigHeaderArray(TClonesArray* array) {mTrigHeaderArrayRef = array;}
+  void setTrigHeaderArray(TRef arrayRef) {mTrigHeaderArrayRef = arrayRef;}
+
+  //these require vertices clones array to be filled first
+  void setBestVert(const StJetSkimVert & v);
+  void setBestVert(int clonesArrayIndex);
+  
+  void setFill(float i) {mFill = i;}
+  void setRunId(int i) {mRunId = i;}
+  void setEventId(int i) {mEventId = i;}
+  void setMudstFileName(const TObjString& i) {mMudstFileName = i;}
+  
+  void setBx7(int i) {mbx7 = i;}
+  void setBx48(int i) {mbx48 = i;}
+  void setSpinBits(int i) {mSpinBits = i;}    
+  
+  void setEbbc(int i) {mEbbc = i;}
+  void setWbbc(int i) {mWbbc = i;}
+  void setBbcTimeBin(int i) {mBbcTimeBin = i;}
+
+  void setVpdTdiff  (float x) {mVpdTdiff   = x;}
+  void setVpdTstart (float x) {mVpdTstart  = x;}
+  void setVpdZvertex(float x) {mVpdZvertex = x;}
+  void setVpdEastHits(int i) {mVpdEastHits = i;}
+  void setVpdWestHits(int i) {mVpdWestHits = i;}
+
+  void setZdcWestRate(float x) { mZdcWestRate = x; }
+  void setZdcEastRate(float x) { mZdcEastRate = x; }
+  void setZdcCoincidenceRate(float x) { mZdcCoincidenceRate = x; }
+
+  void setBbcWestRate(float x) { mBbcWestRate = x; }
+  void setBbcEastRate(float x) { mBbcEastRate = x; }
+  void setBbcCoincidenceRate(float x) { mBbcCoincidenceRate = x; }
+
+  void setIsValid(int i) {mIsValid = i;}
+  void setIsPolLong(int i) {mIsPolLong = i;}
+  void setIsPolTrans(int i) {mIsPolTrans = i;}
+  void setIsMaskedUsingBx48(int i) {mIsMaskedUsingBx48 = i;}
+  void setOffsetBx48minusBX7(int i) {mOffsetBx48minusBX7 = i;}
+  void setSpin4UsingBx48(int i) {mSpin4usingBx48 = i;}
+  void setL2Result(const int* rhs);
+  void setMcEvent(const StPythiaEvent *ptr) {mMcEvent = ptr;}
+
+  void setNTzero(short n) { mNTzero = n; }
+  void setNTzeroCan(short nCan) { mNTzeroCan = nCan; }
+  void setTCanFirst(double tFirst) { mTCanFirst = tFirst; }
+  void setTCanLast(double tLast) { mTCanLast = tLast; }
```

Comment:
Would be cool if someone knowledgeable would add a comment on what those values are.
I could not find them https://www.star.bnl.gov/webdata/dox/html/classStBTofHeader.html

Reply:
Hi @zlchang and @veprbl! Since I can't make the edits to this file here are the comments that can be added for these variables:

mNTzero: Number of particles used for start-time calculations
mNTzeroCan: Number of candidate particles available for start-time calculations before outlier rejection
mTCanFirst: Earliest candidate particle in start-time calculations
mTCanLast: Latest candidate particle in start-time calculations

Please, let me know if you would like more details.

Reply:
@BassamAboona Thanks!

```suggestion
  void setNTzero(short n) { mNTzero = n; } //! Set number of particles used for start-time calculations
  void setNTzeroCan(short nCan) { mNTzeroCan = nCan; } //! Set number of candidate particles available for start-time calculations before outlier rejection
  void setTCanFirst(double tFirst) { mTCanFirst = tFirst; } //! Set earliest candidate particle in start-time calculations
  void setTCanLast(double tLast) { mTCanLast = tLast; } //! Set latest candidate particle in start-time calculations
```